radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/particles/quarks/Quark.java

```diff
@@ -0,0 +1,7 @@
+package com.radixdlt.client.core.atoms.particles.quarks;
+
+/**
+ * A distinct property of a {@link com.radixdlt.client.core.atoms.particles.Particle}
+ */
+public abstract class Quark {
```

Comment:
This should include:
```
    public abstract Set<Spin> allowableStates();
```
in order to be able to runtime analysis (and potentially static) on which quarks can go together in a particle and which cannot.